engine/base/api/src/main/scala/pl/touk/nussknacker/engine/baseengine/api/runtimecontext/EngineRuntimeContextPreparer.scala

```diff
@@ -19,5 +19,9 @@ class EngineRuntimeContextPreparer(metricRegistryForScenario: String => MetricsP
 
 case class BaseEngineRuntimeContext(jobData: JobData,
                                     metricsProvider: MetricsProviderForScenario with AutoCloseable) extends EngineRuntimeContext with AutoCloseable {
+
+  override def contextIdGenerator(nodeId: String): ContextIdGenerator = new IncContextIdGenerator(jobData.metaData.id + "-" + nodeId)
```

Comment:
I wonder, maybe we can have a constructor for IncContextIdGenerator which takes JobData & nodeId?

Reply:
Added factory method `withProcessIdNodeIdPrefix` - don't want to have `apply` because it is not the only one (trivial) option for this prefix. For flink it can confuse someone and can choose the bad one option (without subtask id)